adapters/repos/db/vector/hnsw/index.go

```diff
@@ -416,25 +416,22 @@ func (h *hnsw) findBestEntrypointForNode(currentMaxLevel, targetLevel int,
 	for level := currentMaxLevel; level > targetLevel; level-- {
 		eps := priorityqueue.NewMin[any](1)
 		var dist float32
-		var ok bool
 		var err error
 		if h.compressed.Load() {
-			dist, ok, err = distancer.DistanceToNode(entryPointID)
-			var e storobj.ErrNotFound
-			if errors.As(err, &e) {
-				h.handleDeletedNode(e.DocID)
-				continue
-			}
+			dist, err = distancer.DistanceToNode(entryPointID)
 		} else {
-			dist, ok, err = h.distBetweenNodeAndVec(entryPointID, nodeVec)
+			dist, err = h.distBetweenNodeAndVec(entryPointID, nodeVec)
+		}
+
+		var e storobj.ErrNotFound
+		if errors.As(err, &e) {
+			h.handleDeletedNode(e.DocID)
```

Comment:
this method was only invoked in case of `compressed` check but is now invoked in both, that's ok?

Reply:
i would say so, given that it checks if the docID has tombstone and if not it add it